src/internal/operators/shareReplay.ts

```diff
@@ -5,6 +5,42 @@ import { MonoTypeOperatorFunction, SchedulerLike } from '../types';
 import { Subscriber } from '../Subscriber';
 
 /**
+ * Share source and replay specified number of emissions on subscription.
+ *
+ * This operator is a specialization of `replay` that connects to the connectable observable sequence
+ * when the number of observers goes from zero to one, and disconnects when there are no more observers.
+ *
+ * ## Why use shareReplay?
+ * You generally want to use `shareReplay` when you have side-effects or taxing computations
+ * that you do not wish to be executed amongst multiple subscribers.
+ * It may also be valuable in situations where you know you will have late subscribers to
+ * a stream that need access to previously emitted values.
+ * This ability to replay values on subscription is what differentiates {@link share} and `shareReplay`.
+ *
+ * ![](shareReplay.png)
+ *
+ * ## Example
+ * ```javascript
+ * const obs$ = interval(1000);
+ * const subscription = obs$.pipe(
+ *   take(4),
+ *   shareReplay(3)
+ * );
+ * subscription.subscribe(x => console.log('source A: ', x));
+ * subscription.subscribe(y => console.log('source B: ', y));
+ *
+ * ```
+ *
+ * @see {@link publish}
+ * @see {@link share}
+ * @see {@link publishReplay}
+ *
+ * @param {Number} [bufferSize] Maximum element count of the replay buffer.
```

Comment:
Mention that it defaults to an infinite number of elements.

Reply:
It is quite wired that the unit tests failed in CI after I added the default values 'Number.POSITIVE_INFINITY' to 'bufferSize'.

Reply:
Done.